lib/rules/forbid-component-props.js

```diff
@@ -38,11 +45,31 @@ module.exports = {
 
   create: function(context) {
 
+    function somePatternMatches(patterns, subject) {
+      return patterns.reduce(function (acc, pattern) {
+        if (RegExp(pattern).test(subject)) {
```

Comment:
we're iterating through the array fo patterns, so each pattern is transformed to a regex only once